docs/source/add_new_adaptor.md

```diff
@@ -0,0 +1,243 @@
+    quantizable_op_details = {
+      ('conv2d', 'Conv2D'): [int8_conv_config, bf16_config, fp32_config]
+    }
+
+```
+In most case, `bf16_config` and `fp32_config` stay the same on different operators and we can add config of different data type to the `quantizable_op_details`. It's configured in the YAML file and parsed out with format like below.
+
+```python
+    int8_conv_config = {
+        'weight': {
+               'dtype': 'int8',
+               'algorithm': 'minmax',
+               'granularity': 'per_channel',
+               'scheme': 'sym'
+        },
+        'activation': {
+               'dtype': 'int8',
+               'quant_mode': 'static',
+               'algorithm': 'kl',
+               'granularity': 'per_tensor',
+               'scheme': 'sym'
+        }
+    }
+
+```
+After we got the opwise capability of the model, we should also get the optypewise ability. It's also a key-value pair but the key is op type and the value is the configuration of activation and weight for this op type. It has format like below:
+
+```python
+    optype_wise_ability = {
+        'Conv2D': {
+            'weight': {
+                   'dtype': 'int8',
+                   'algorithm': 'minmax',
+                   'granularity': 'per_channel',
+                   'scheme': 'sym'
+            },
+            'activation': {
+                   'dtype': 'int8',
+                   'quant_mode': 'static',
+                   'algorithm': 'kl',
+                   'granularity': 'per_tensor',
+                   'scheme': 'sym'
+            }
+        }
+    }
+```
+There is an API get_optype_wise_ability to get the optype_wise_ability from the `quantizable_op_details`. You can refer to [get_optype_wise_ability](../neural_compressor/adaptor/keras.py#L614) for the implementation.
+
+After the work above, we have implement the `query_fw_capability` API and get the tuning capability dict for the Strategy object. Then the Strategy object will fetch tuning configuration and give to the quantize API to get the quantized model.
+
+## Add Pre-quantization optimizations
+As the `query_fw_capability` only invoked once, we can do some pre-optimizations in the `query_fw_capability` API and cache the pre_optimized_object for quantization. It's not mandatory and if you have done optimizations on the model, take care to use the optimized model in the quantize API.
+
+
+## Quantize Model according to tune_cfg
+`quantize` function is used to perform quantization for post-training quantization and quantization-aware training. Quantization processing includes calibration and conversion processing for post-training quantization, while for quantization-aware training, it includes training and conversion processing.
+
+The first work of `quantize` function is to invoke `tuning_cfg_to_fw` to generate the self.quantize_config. The self.quantize_config is a dict including the quantization information. It's format is like below
+
+```
+self.quantize_config  = {
```

Comment:
maybe integrate with working flow is better? I think here is used to describe what's the input format of the quantize() func in adaptor, right?